### PR TITLE
Use Mutiny boms instead of listing all dependencies individually

### PR DESCRIPTION
**Pull request by cescoffier**
Use the Mutiny BOMs instead of listing the dependencies individually.
There are two boms:

* the mutiny bom - Mutiny API and related projects
* the Mutiny Vert.x Bindings - Provides the Vert.x Mutiny APIs (clients, core...)

### CODE REVIEW

**Review by jponge — APPROVED** [review]

File: bom/application/pom.xml [hunk 1]
```diff
@@ -4559,78 +4541,12 @@
...
-            </dependency>
-            <dependency>
-                <groupId>io.smallrye.reactive</groupId>
-                <artifactId>smallrye-mutiny-vertx-redis-client</artifactId>
-                <version>${smallrye-reactive-utils.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.vertx</groupId>
-                <artifactId>vertx-kafka-client</artifactId>
-                <version>${vertx.version}</version>
```
Comment (gsmet) [inline bom/application/pom.xml:4606]: What about this one? It was unnecessary?

**Review by gsmet — DISMISSED** [review]
Asked a question. Would prefer clarification before merging. Feel free to dismiss once done.

File: bom/application/pom.xml (hunk 1 above)
Comment (cescoffier) [reply bom/application/pom.xml:4606]: Yes, we are not using it anymore.

Comment (cescoffier) [reply bom/application/pom.xml:4606]: Actually, we still have a dependency - so better keep it. 
We are going to get rid of it, but one step at a time.

**Review by geoand — APPROVED** [review]

**Pull request merged**